mods/classic-file-picker-dialog.wh.cpp

```diff
@@ -0,0 +1,67 @@
+// ==WindhawkMod==
+// @id              classic-file-picker-dialog
+// @name            Classic File Picker dialog
+// @description     Redirect the Windows Vista+ file picker to the Windows XP one
+// @version         1.0
+// @author 	        Anixx
+// @github          https://github.com/Anixx
+// @include         *
+// ==/WindhawkMod==
+
+// ==WindhawkModReadme==
+/*
+This mod replaces the default file picker (open/save dialog) with the legacy one,
+which was used in Windows XP and Windws 2000.
+
+For 3D border in the dialog, install the mod "ClientEdge in choosen apps"
+
+The mod is based on the idea by xdmg01.
+
+![File Picker](https://i.imgur.com/g2NVokJ.png)
+
+*/
+// ==/WindhawkModReadme==
+
+#include <windhawk_utils.h>
+
+#ifdef _WIN64
+#define CALCON __cdecl
+#define SCALCON L" __cdecl "
+#else
+#define CALCON __thiscall
+#define SCALCON L" __thiscall "
+#endif
+
+typedef int (CALCON *IsCommonItemDialogAvailable)(void);
+IsCommonItemDialogAvailable IsCommonItemDialogAvailable_orig;
+
+int CALCON IsCommonItemDialogAvailable_hook(void) {return 0;}
+
+typedef long (CALCON *CFileOpenSave__s_CreateInstance)(int, GUID*, void**);
+CFileOpenSave__s_CreateInstance CFileOpenSave__s_CreateInstance_orig;
+
+long CALCON CFileOpenSave__s_CreateInstance_hook(int param_1, GUID *param_2, void **param_3) {
+    param_1 = (param_1 > 1) ? param_1 -= 2 : param_1;
+    return CFileOpenSave__s_CreateInstance_orig(param_1, param_2, param_3);
+}
+
+BOOL Wh_ModInit() {
+    HMODULE module = LoadLibrary(L"comdlg32.dll");
+
+    WindhawkUtils::SYMBOL_HOOK symbolHook[] = {
+        {
+            {L"int" SCALCON L"IsCommonItemDialogAvailable(void)"},
```

Comment:
Are you sure it should be `__thiscall` for 32-bit? I checked the copies of `comdlg32.dll`, and the relevant symbol is:
```
int __stdcall IsCommonItemDialogAvailable(void)
```

Same for the second symbol below.